parlai/core/torch_generator_agent.py

```diff
@@ -915,6 +951,18 @@ def advance(self, logprobs):
                     if prefix == list(ngram[:-1]) or self.block_ngram == 1:
                         logprobs[beam_id][ngram[-1]] = neginf(logprobs.dtype)
 
+        if self.context_block_ngram > 0:
+            if self.context is None:
+                raise ValueError(
+                    "Must use TreeSearch.set_context to use context blocking."
+                )
+            for beam_id, hyp in enumerate(self.partial_hyps):
```

Comment:
pretty nitty comment but as this code is nearly identical to one above, we could just have a function `suppress_ngram_logprob` that executes these 5 lines with the appropriate values (just saves some space 😄)

Reply:
actually, no we can't do that, because the ngram length differs, so prefix differs, which makes the refactor more cumbersome than the original.

Reply:
maybe not, idk, i'll give it a shot

Reply:
Okay, i think you'll like this better